lib/app/service/docker_test.go

```diff
@@ -82,3 +84,63 @@ func (s *VendorSuite) TestGeneratesProperPackageName(c *C) {
 		c.Assert(*runtimePackage, compare.DeepEquals, testCase.result, comment)
 	}
 }
+
+func Test_excludeImagesStartingWith(t *testing.T) {
+	type args struct {
+		images []string
+		s      string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "empty slice",
+			args: args{
+				images: []string{},
+				s:      "123",
+			},
+			want: []string{},
+		},
+		{
+			name: "empty pattern",
+			args: args{
+				images: []string{"name1", "name2"},
+				s:      "",
+			},
+			want: []string{"name1", "name2"},
+		},
+		{
+			name: "exclude all images. result is empty slice",
+			args: args{
+				images: []string{"registry1/name1", "registry1/name2"},
+				s:      "registry1",
+			},
+			want: []string{},
+		},
+		{
+			name: "exclude images. result is not empty",
+			args: args{
+				images: []string{"registry1/name1", "registry1/name2", "registry2/name2", "registry2/name3"},
+				s:      "registry2",
+			},
+			want: []string{"registry1/name1", "registry1/name2"},
+		},
+		{
+			name: "doesn't exclude images",
+			args: args{
+				images: []string{"registry1/name1", "registry1/name2", "registry2/name2"},
+				s:      "registry3",
+			},
+			want: []string{"registry1/name1", "registry1/name2", "registry2/name2"},
+		},
```

Comment:
I would add tests that mix the pattern - i.e. you have pattern in the beginning test but but no pattern in the middle and pattern as a suffix.